packages/manager/apps/pci-savings-plan/src/utils/formatter/formatter.ts

```diff
@@ -58,3 +60,62 @@ export const formatPricingInfo = (
     return {};
   }
 };
+
+type DayData = {
+  day: number;
+  included: number;
+  excluded: number;
+  cumulPlanSize: number;
+};
+
+const filterMaxIncludedPerDay = (data: DayData[]): DayData[] => {
+  const maxIncludedMap: Record<number, DayData> = {};
+
+  data.forEach((entry) => {
+    const existingEntry = maxIncludedMap[entry.day];
+    if (!existingEntry || entry.included > existingEntry.included) {
+      maxIncludedMap[entry.day] = entry;
+    }
+  });
+
+  return Object.values(maxIncludedMap);
+};
+
+export const getChartsData = (
+  periods: SavingsPlanPeriodConsumption[],
+): DayData[] =>
+  filterMaxIncludedPerDay(
+    periods
+      .map((period) => {
+        const startDate = period.begin;
+        const endDate = period.end;
+        const computedEndDate = sub(endDate, { minutes: 1 }).toISOString();
+        const adjustedEndDate = new Date(computedEndDate);
+        adjustedEndDate.setUTCHours(0, 0, 0, 0);
+
+        const consumptionSizeFormatted = period.consumption_size ?? 0;
+        const cumulPlanSizeFormatted = period.cumul_plan_size ?? 0;
+
+        const included = Math.min(
+          consumptionSizeFormatted,
+          cumulPlanSizeFormatted,
+        );
+        const excluded =
+          consumptionSizeFormatted > cumulPlanSizeFormatted
+            ? consumptionSizeFormatted - cumulPlanSizeFormatted
+            : 0;
+
+        const daysInterval = eachDayOfInterval({
+          start: startDate,
+          end: adjustedEndDate,
+        });
+
+        return daysInterval.map((day) => ({
+          day: day.getDate(),
+          included,
+          excluded,
+          cumulPlanSize: cumulPlanSizeFormatted,
+        }));
+      })
+      .flatMap((period) => period),
```

Comment:
```suggestion
      .flat(),
```

Reply:
I see map(()=>{}).flat().
I think directly flatMap(()=>{}) can do the same thing